src/gc/impl/conservative/gc.d

```diff
@@ -1346,6 +1346,12 @@ struct Gcx
         toscanPrecise.reset();
     }
 
+    __gshared bool inShutdown = false;
+    shared static ~this()
+    {
+        // detect shutdown to avoid starting threads for parallel marking
+        inShutdown = true;
+    }
```

Comment:
This works, but why not just tell the GC it's being shut down here?

https://github.com/dlang/druntime/blob/0f9cd780759765241e57fdb0e594ae61798d7e46/src/gc/proxy.d#L120

Reply:
Indeed, that is a simpler check I didn't think of. I was also slightly worried about introducing module dependencies due to the static dtor. I'll change it.